interfaces/builtin/maliit_input_method_test.go

```diff
@@ -0,0 +1,209 @@
+			},
+			Name:      "maliit-input-method",
+			Interface: "maliit-input-method",
+			Apps:      map[string]*snap.AppInfo{"app1": app1, "app2": app2},
+		},
+	}
+	snippet, err := s.iface.ConnectedSlotSnippet(plug, s.slot, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	c.Assert(string(snippet), testutil.Contains, `peer=(label="snap.maliit-input-method.*"),`)
+}
+
+// The label uses alternation when some, but not all, apps is bound to the maliit-input-method plug
+func (s *MaliitInputMethodInterfaceSuite) TestConnectedSlotSnippetUsesPlugLabelSome(c *C) {
+	app1 := &snap.AppInfo{Name: "app1"}
+	app2 := &snap.AppInfo{Name: "app2"}
+	app3 := &snap.AppInfo{Name: "app3"}
+	plug := &interfaces.Plug{
+		PlugInfo: &snap.PlugInfo{
+			Snap: &snap.Info{
+				SuggestedName: "maliit-input-method",
+				Apps:          map[string]*snap.AppInfo{"app1": app1, "app2": app2, "app3": app3},
+			},
+			Name:      "maliit-input-method",
+			Interface: "maliit-input-method",
+			Apps:      map[string]*snap.AppInfo{"app1": app1, "app2": app2},
+		},
+	}
+	snippet, err := s.iface.ConnectedSlotSnippet(plug, s.slot, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	c.Assert(string(snippet), testutil.Contains, `peer=(label="snap.maliit-input-method.{app1,app2}"),`)
+}
+
+// The label uses short form when exactly one app is bound to the maliit-input-method plug
+func (s *MaliitInputMethodInterfaceSuite) TestConnectedSlotSnippetUsesPlugLabelOne(c *C) {
+	app := &snap.AppInfo{Name: "app"}
+	plug := &interfaces.Plug{
+		PlugInfo: &snap.PlugInfo{
+			Snap: &snap.Info{
+				SuggestedName: "maliit-input-method",
+				Apps:          map[string]*snap.AppInfo{"app": app},
+			},
+			Name:      "maliit-input-method",
+			Interface: "maliit-input-method",
+			Apps:      map[string]*snap.AppInfo{"app": app},
+		},
+	}
+	snippet, err := s.iface.ConnectedSlotSnippet(plug, s.slot, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	c.Assert(string(snippet), testutil.Contains, `peer=(label="snap.maliit-input-method.app"),`)
+}
+
+func (s *MaliitInputMethodInterfaceSuite) TestPermanentSlotSecComp(c *C) {
+	snippet, err := s.iface.PermanentSlotSnippet(s.slot, interfaces.SecuritySecComp)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, Not(IsNil))
+
+	c.Check(string(snippet), testutil.Contains, "listen\n")
+}
+
+func (s *MaliitInputMethodInterfaceSuite) TestUsedSecuritySystems(c *C) {
+	systems := [...]interfaces.SecuritySystem{interfaces.SecurityAppArmor,
+		interfaces.SecuritySecComp}
+	for _, system := range systems {
+		snippet, err := s.iface.ConnectedPlugSnippet(s.plug, s.slot, system)
+		c.Assert(err, IsNil)
+		c.Assert(snippet, Not(IsNil))
+		snippet, err = s.iface.PermanentSlotSnippet(s.slot, system)
+		c.Assert(err, IsNil)
+		c.Assert(snippet, Not(IsNil))
+	}
+	snippet, err := s.iface.ConnectedSlotSnippet(s.plug, s.slot, interfaces.SecurityAppArmor)
+	c.Assert(err, IsNil)
+	c.Assert(snippet, Not(IsNil))
+}
```

Comment:
The way TestUsedSecuritySystems is written is the old way. Please add separate tests:
- TestConnectedPlugSnippetAppArmor
- TestConnectedPlugSnippetSecComp
- TestConnectedSlotSnippetAppArmor
- TestPermanentSlotSnippetAppArmor
- TestPermanentSlotSnippetSecComp

and have each check for NotNil as well as for an expected string in the policy. See upower_observe_test.go for inspiration.